libraries/package-extractor/src/PackageExtractor.ts

```diff
@@ -646,6 +698,31 @@ export class PackageExtractor {
       // and there is no exclude filter, then we know that the file is not excluded. If it is included
       // and there is an exclude filter, then we need to check for a match.
       return !isIncluded || !!excludeFilters?.some((m) => m.match(filePath));
+    };
+    // Function to filter files inside local project
+    const isFileExcludedForProject = (filePath: string): boolean =>
+      isFileExcluded(filePath, {
+        patternsToExclude: sourceProjectConfiguration?.patternsToExclude,
+        patternsToInclude: sourceProjectConfiguration?.patternsToInclude
+      });
+    // Function to filter files inside third party dependencies
+    const isFileExcludedForDependency = (filePath: string): boolean => {
+      if (!packagesJson) {
+        return false;
+      }
+      const dependenciesConfiguration: IExtractorDependencyConfiguration | undefined =
+        dependencyConfigurationsByNameAndVersion.get(`${packagesJson.name}@${packagesJson.version}`);
+      if (!dependenciesConfiguration) {
+        return false;
+      }
+
+      return isFileExcluded(filePath, dependenciesConfiguration);
+    };
+    if (sourceProjectConfiguration) {
+      if (!includeNpmIgnoreFiles) {
```

Comment:
Combine this with the outer if statement

Reply:
Better yet, just set the property to `sourceProjectConfiguration && !includeNpmIgnoreFiles`